API/Controllers/OAuthController.cs

```diff
@@ -74,4 +76,20 @@ public class OAuthController(IOAuthHandler oAuthHandler) : Controller
 
         return TypedResults.Ok(result);
     }
+
+    [HttpPost("ratelimit")]
+    [Authorize(Roles = "admin")]
+    [EndpointDescription("Sets new ratelimit overrides for a given client")]
+    public async Task<Results<BadRequest, Ok<OAuthClientDTO>>> SetRatelimitOverridesAsync([FromQuery] int clientId, [FromQuery] int window, [FromQuery] int limit)
+    {
+        OAuthClientDTO? result = await clientService.SetRatelimitOverridesAsync(clientId,
+            new RateLimitOverrides { PermitLimit = limit, Window = window });
+
+        if (result == null)
+        {
+            return TypedResults.BadRequest();
```

Comment:
```suggestion
            return TypedResults.NotFound();
```
Make sure return type matches